### PR TITLE
SQSERVICES-559 make `AllFeatureConfigs` typed

### PR DESCRIPTION
**Pull request by battermann**
https://wearezeta.atlassian.net/browse/SQSERVICES-1559

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: libs/wire-api/src/Wire/API/Team/Feature.hs [hunk 1]
```diff
@@ -362,6 +360,59 @@ modelForTeamFeature TeamFeatureGuestLinks = modelTeamFeatureStatusNoConfig
...
+    afcFileSharingInternal :: TeamFeatureStatus 'WithLockStatus 'TeamFeatureFileSharing,
+    afcClassifiedDomainsInternal :: TeamFeatureStatus 'WithoutLockStatus 'TeamFeatureClassifiedDomains,
+    afcConferenceCallingInternal :: TeamFeatureStatus 'WithoutLockStatus 'TeamFeatureConferenceCalling,
+    afcSelfDeletingMessagesInternal :: TeamFeatureStatus 'WithLockStatus 'TeamFeatureSelfDeletingMessages,
+    afcGuestLinkInternal :: TeamFeatureStatus 'WithLockStatus 'TeamFeatureGuestLinks,
+    afcSndFactorPasswordChallengeInternal :: TeamFeatureStatus 'WithLockStatus 'TeamFeatureSndFactorPasswordChallenge
+  }
+  deriving stock (Eq, Show)
+  deriving (FromJSON, ToJSON, S.ToSchema) via (Schema AllFeatureConfigs)
+
```
Comment (fisx) [inline libs/wire-api/src/Wire/API/Team/Feature.hs:379]:
```suggestion
  deriving (Arbitrary) via (GenericUniform AllFeatureConfigs)

```

(not tested!  but if it works, you can drop the implicit implementation below.)

**Review by fisx — COMMENTED** [review]
almost done reading, but i have to go, sorry~

File: libs/wire-api/src/Wire/API/Team/Feature.hs (hunk 1 above)
Comment (battermann) [reply libs/wire-api/src/Wire/API/Team/Feature.hs:379]: can't get it to wrok

File: services/galley/src/Galley/API/Teams/Features.hs [hunk 2]
```diff
@@ -563,11 +546,17 @@ setFileSharingInternal = Tagged $ \tid status -> do
...
+  FeatureScopeTeam tid -> do
+    cfgDefault <- getCfgDefault
+    mStatus <- TeamFeatures.getApplockFeatureStatus tid
+    pure $ fromMaybe cfgDefault mStatus
+  FeatureScopeUser _ -> getCfgDefault
+  FeatureScopeServer -> getCfgDefault
+  where
+    getCfgDefault = do
+      Defaults defaultStatus <- inputs (view (optSettings . setFeatureFlags . flagAppLockDefaults))
+      pure defaultStatus
```
Comment (fisx) [inline services/galley/src/Galley/API/Teams/Features.hs:559]:
```suggestion
      inputs (view (optSettings . setFeatureFlags . flagAppLockDefaults . unDefaults))
```

**Review by fisx — APPROVED** [review]

**Pull request merged**